docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_direct_internet_access.mdx

```diff
@@ -23,6 +23,277 @@ GDPR, HITRUST, SOC2
+
+   ```
+   aws sagemaker describe-notebook-instance --notebook-instance-name <YourNotebookInstanceName>
+   ```
+   Replace `<YourNotebookInstanceName>` with the name of your SageMaker notebook instance. This command will return a JSON object that contains detailed information about the notebook instance, including the `DirectInternetAccess` attribute. If the value of this attribute is `Enabled`, it means that the notebook instance has direct internet access.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check for Notebook Direct Internet Access in DynamoDB using Python scripts, you can use the Boto3 library, which allows you to directly create, update, and delete AWS services from your Python scripts. Here are the steps:
+
+1. **Import the Boto3 library in Python:**
+   Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then import it in your script:
+   ```python
+   import boto3
+   ```
+
+2. **Create a DynamoDB resource and client:**
+   You can create a DynamoDB resource and client using your AWS credentials. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials.
+   ```python
+   dynamodb = boto3.resource('dynamodb', aws_access_key_id='your_access_key', aws_secret_access_key='your_secret_key', region_name='your_region')
+   client = boto3.client('dynamodb', aws_access_key_id='your_access_key', aws_secret_access_key='your_secret_key', region_name='your_region')
+   ```
+
+3. **List all the DynamoDB tables:**
+   You can list all your DynamoDB tables using the `list_tables` method.
+   ```python
+   response = client.list_tables()
+   print(response['TableNames'])
+   ```
+
+4. **Check for Notebook Direct Internet Access:**
+   DynamoDB doesn't directly relate to Notebook Direct Internet Access. However, if you're referring to checking if a DynamoDB table is publicly accessible, you can do so by checking the policy attached to the DynamoDB table. You can use the `get_item` method to retrieve the policy and then parse the policy to check if it allows public access.
+   ```python
+   for table_name in response['TableNames']:
+       policy = client.get_item(TableName=table_name, Key={'AttributeName': 'Policy'})['Item']
+       if 'Allow' in policy['Statement'][0]['Effect'] and 'Principal' in policy['Statement'][0] and policy['Statement'][0]['Principal'] == '*':
+           print(f'Table {table_name} is publicly accessible')
+   ```
+   Please note that this is a simplified example and actual policy parsing might be more complex depending on the policy structure.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>